engine/glfw/lib/android/android_window.c

```diff
@@ -442,17 +442,21 @@ void glfwAndroidPollEvents()
    int events;
    struct android_poll_source* source;
 
-   int timeout = 0;
+   int timeoutMillis = 0;
    if (_glfwWin.iconified) {
-       timeout = 1000 * 3000;
+        timeoutMillis = 1000 * 3;
    }
-   while ((ident=ALooper_pollAll(timeout, NULL, &events, (void**)&source)) >= 0)
+   while ((ident=ALooper_pollAll(timeoutMillis, NULL, &events, (void**)&source)) >= 0)
    {
-       timeout = 0;
-       // Process this event.
-       if (source != NULL) {
+        timeoutMillis = 0;
+        // Process this event.
+        if (source != NULL) {
            source->process(_glfwWinAndroid.app, source);
-       }
+        }
+
+        if (app->destroyRequested != 0) {
+            return;
```

Comment:
If app destroy requested we can exit the loop

Reply:
Ok, and this is set from android_native_app_glue.c?

Reply:
[yes](https://developer.android.com/reference/games/game-activity/struct/android-app#destroyrequested), `android_native_app_glue.c` rise this flag and then waits when app destroys (exit from `android_main()` in our case this is `engine_main()`)

But I'm not sure If should directly call return here or maybe it would be better to set `_glfwWin.opened = 0;` here and make sure we exit app (even if we didn't get `APP_CMD_DESTROY` yet. Because if we exit this loop it's possible we will not get it)

Reply:
Since we've had issues with OpenAL and shutting down, won't it cause more issues if we simply exit here? If not, then I guess it's safe to exit here.

If we should wait for the game/sound threads, then I think shutting down sooner is preferred. So somehow make sure the engine starts the shutdown sequence?